google/ggrc-core
# GGRC-4774 Mapped objects to Cycle Task get unmapped after saving Cycle Task

# Issue description

Mapped objects to Cycle Task get unmapped after saving Cycle Task.

# Steps to test the changes

Steps to reproduce:
1. Open My work page > Tasks tab
2. Create a new cycle task and map any object to Cycle task, e.g. Control
3. Open Edit Cycle task popup and unmap Control
4. Expand Cycle task Info panel and map Control again
5. Open Edit Cycle task popup and click Save and Close
6. Look at the Cycle task Info panel: mapped Control is not displayed
*Actual Result:* Mapped objects to Cycle Task get unmapped after saving Cycle Task
*Expected Result:* Mapped objects to Cycle Task should not be unmapped after saving Cycle Task

# Solution description

This issue happened due to incorrect `destroy` handler within `ggrc-modal-connector` controller. When someone closes a modal with `ggrc-modal-connector` component, then it should be destroyed. Each CanJs's component has ability to catch `destroyed` event. But in our case, `ggrc-modal-connector` has `destroy` (not `destroyed`) handler, which overrides `can.Control`'s `destroy` method (from docs:  **A component's events object is used as the prototype of a can.Control**). Thus, we'll use undestroyed parts of this component in the next modal calls.
Solution: `ggrc-modal-connector`'s code was analyzed and refactored. During the refactoring several strange pieces of code were removed, including `destroy` handler.

For small help:

https://github.com/google/ggrc-core/commit/f0f7a14c49d6b0631bec173deeccd6f9d83218e1#diff-2f5b1b7c5a393e85330925ab5a3ec3b7 - Remove this `CMS.ModalHelpers` object.


https://github.com/google/ggrc-core/commit/7d307cede18cf002a3b65b519d58a3561f41aaba - Remove unused helpers + `options.instance`

# Sanity checklist




# PR Review checklist
